patterns/choreography-to-orchestration.md

```diff
@@ -0,0 +1,59 @@
+# Convert Choreography to Orchestration
+
+![](images/choreography-to-orchestration/choreography.png)
+
+***
+
+![](images/choreography-to-orchestration/orchestration.png)
+
+***
+
+## Description:
+
+Choreography and Orchestration are two modes on how distributed microservices services can communicate with each another.
+
+In Choreography, every service works independently and achieves communication without tight control. Events flow between services without any centralized coordination. Each microservice listens for events of it's interest and does its own thing, following the event-driven paradigm
+
+In orchestration, there is a central service (the ‘orchestrator’) that controls the interaction between services. It coordinates the interaction and order in which services are invoked like a control flow for the business logic. 
+
+Many applications use both choreography and orchestration for different use cases. Communication between bounded contexts is often how choreography is used most effectively, within a bounded context, you need to control the sequence of service integration, maintain state, and handle errors and retries. A standard guideline is to use orchestration within the bounded context of a microservice, but use choreography between bounded-contexts. 
```

Comment:
So that reader can understand what a bounded context is, I think you could add link to its definition. Example: https://martinfowler.com/bliki/BoundedContext.html

Reply:
Added hyperlinks